docs/userguide/constraints.md

```diff
@@ -221,39 +241,162 @@ DiscreteLinkedParametersConstraint(
+`SearchSpace` will still retain **one** configuration for that setting, showing arbitrary
+values for the `affected_parameters` (which can be ignored).
+
+(DDC)=
+```{note}
+BayBE requires that all dependencies are declared in a single
+`DiscreteDependenciesConstraint`. Creating a `SearchSpace` from multiple
+`DiscreteDependenciesConstraint`'s will throw a validation error.
+```
+
+In the example below, we mimic a situation where there are two switches and each switch
+activates two other parameters that are only relevant if the first switch is `"on"` / the
+second switch is set to `"right"`, respectively.
+```python
+from baybe.constraints import DiscreteDependenciesConstraint, SubSelectionCondition
+
+DiscreteDependenciesConstraint(
+    parameters=["Switch_1", "Switch_2"],  # the two parameters upon which others depend
+    conditions=[
+        SubSelectionCondition(selection=["on"]),     # values of Switch_1 that activate the affected parameters
+        SubSelectionCondition(selection=["right"]),  # values of Switch_2 that activate the affected parameters
+    ],
+    affected_parameters=[
+      ["Solvent", "Fraction"],  # parameters affected by Switch_1
+      ["Frame_1", "Frame_2"]    # parameters affected by Switch_2
+    ],
+)
+```
+
+An end to end example can be found [here](../../examples/Constraints_Discrete/dependency_constraints).
+
+### DiscretePermutationInvarianceConstraint
+Permutation invariance, enabled by the 
+[`DiscretePermutationInvarianceConstraint`](baybe.constraints.discrete.DiscretePermutationInvarianceConstraint)
+, is a property where combinations of values of multiple
+parameters do not depend on their order due to some symmetry in the experiment.
+Suppose we create a mixture containing up to three solvents, i.e. parameters
+"Solvent_1", "Solvent_2", "Solvent_3".
+In this situation, all combinations from the following table would be equivalent,
+hence the `SearchSpace` should effectively only contain one of them.
+
+|   | Solvent_1    | Solvent_2    | Solvent_3    |
+|---|--------------|--------------|--------------|
+| 1 | Substance_43 | Substance_3  | Substance_12 |
+| 2 | Substance_43 | Substance_12 | Substance_3  |
+| 3 | Substance_3  | Substance_12 | Substance_43 |
+| 4 | Substance_3  | Substance_43 | Substance_12 |
+| 5 | Substance_12 | Substance_43 | Substance_3  |
+| 6 | Substance_12 | Substance_3  | Substance_43 |
 
-### ``DiscretePermutationInvarianceConstraint``
-Content coming soon...
+```{note}
+Complex properties such as permutation invariance not only affect the search space but
+should ideally also constrain the surrogate model. For instance, the kernels in a
+Gaussian process can be made permutation-invariant to reflect this constraint, which
+generally results in a better learning curve. Note that at this stage no
+surrogate model provided by BayBE takes care of these invariances. This means the
+invariance is ignored during model fitting. This just means these models cannot benefit
```

Comment:
Repeating: "This means ..... this just means ..."

Reply:
rephrased